solutions/silver/cses-1630.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: cses-1630
+source: cses
+title: Tasks & Deadlines
+author: Rameez Parwez
+---
+
+[Unofficial Analysis](https://usaco.guide/CPH.pdf#p=70)
+
+## Explanation
+
+Notice that the meta is to process shorter tasks before longer tasks.
+
+For some intuition regarding this, think of two tasks that take $5$ and $7$ time units to complete.
+No matter what their respective rewards are, it's always optimal to complete the shorter task first.
```

Comment:
It would be better to explain a bit more about why it's optimal. For example, you can mention that sum(d_i - f_i) = sum(d_i) - sum(f_i) where f_i is the finishing time. Since sum(d_i) is constant, f_i is the prefix sum of the duration we process the tasks, and we're trying to minimize sum(f_i), taking a shorter task is optimal.

Reply:
Is it really necessary?
We could at least expect this much from the user to figure it out themselves,  rather than serving everything on the platter.

Reply:
no yeah this is good
if they’re reading the solution it means they’ve given up lol

Reply:
should  we make the test case explanation a spoiler part ?

Reply:
I suppose it is fine I guess. Just though that it'd be useful for people to understand why it's optimal through a proof instead of relying on intuition.

Yeah the example looks good

Reply:
Thanks